src/config/configuration.rs

```diff
@@ -20,22 +20,25 @@ use crate::config::transport_strategy::TransportStrategy;
 /// use momento::config::grpc_configuration::{GrpcConfiguration, GrpcConfigurationBuilder};
 /// use momento::config::transport_strategy::TransportStrategy;
 ///
-/// let config = Configuration::builder(
-///            TransportStrategy::builder(
-///                GrpcConfiguration::builder(Duration::from_millis(1000)).build(),
-///            )
-///            .build(),
-///        )
-///        .build();
-#[derive(Clone)]
+/// let config = Configuration::builder()
+///     .transport_strategy(
+///         TransportStrategy::builder()
+///             .grpc_configuration(
+///                 GrpcConfiguration::builder()
+///                     .deadline(Duration::from_millis(1000))
+///                     .build()    
+///             ).build()
```

Comment:
`grpc_configuration()` should take an `impl Into<GrpcConfiguration>` and `GrpcConfigurationBuilder` should have From impl'd at the appropriate stage.

```
impl From<GrpcConfigurationBuilder<ReadyToBuild>> for GrpcConfiguration {
    fn from(self) -> GrpcConfiguration {
        self.build()
    }
}
```

so that the usage becomes
```
.grpc_configuration(
    GrpcConfiguration::builder()
        .deadline(Duration::from_millis(1000))
).build()
```

And then the transport_strategy function should take an `impl Into<TransportStrategy>`, and then there should be impl From<TransportStrategyBuilder<ReadyToBuild>> for TransportStrategy [...] which results in:

```
/// let config = Configuration::builder()
///     .transport_strategy(
///         TransportStrategy::builder()
///             .grpc_configuration(
///                 GrpcConfiguration::builder().deadline(Duration::from_millis(1000)
///             )
///     )
///     .build();
```

And if you accept `impl Into<Configuration>` in your client constructor then you can simplify most use cases to

```
/// let config = Configuration::builder()
///     .transport_strategy(
///         TransportStrategy::builder()
///             .grpc_configuration(
///                 GrpcConfiguration::builder().deadline(Duration::from_millis(1000)
///             )
///     );
```

Reply:
nice, will try this.

Reply:
fixed in https://github.com/momentohq/client-sdk-rust/pull/191/commits/44fb3db09595c3ee6b41fc20aebfa954198cdaaf